ecl/hql/hqlparse.cpp

```diff
@@ -604,7 +605,24 @@ void HqlLex::checkSignature()
             }
         }
         else
+        {
             yyParser->inSignedModule = true;
+            yyParser->gpgSignature.clear();
+            const char sigprefix[] = "Good signature from \"";
+            const char * const s = buf.str();
+            const char * match = strstr(s, sigprefix);
+            if (match)
+            {
+                match += sizeof(sigprefix) - 1;
+                const char * const end = strchr(match, '\"');
+                if (end)
+                {
+                    buf.setLength(end-s);
+                    const char * sig = buf.str() + (match-s);
+                    yyParser->gpgSignature.setown(createExprAttribute(_fileAccessSigned_Atom,createConstant(sig)) );
```

Comment:
picky: Should probably just be _signed_Atom.  This could also be used for adding signed information to other things.